src/Mod/PartDesign/App/FeatureHole.cpp

```diff
@@ -1664,6 +1664,10 @@ App::DocumentObjectExecReturn* Hole::execute()
     TopoDS_Shape base;
     try {
         base = getBaseShape();
+        if (Passthrough.getValue()) {
+            this->Shape.setValue(base);
+            return App::DocumentObject::StdReturn;
+        }

```

Comment:
This gets repeated over and over again - is there a better place for this piece of code so it does not have to be duplicated? That way feature developer need to remember to cover this case which is common for every feature and thus should be higher in the hierarchy.

I am not sure however if this is possible at all now as this probably would have to be done on the caller side?

Reply:
`execute()` is called by `DocumentObject::recompute()` It's not a PD specific procedure so we can't tweak this.

Reply:
Yeah, I checked how "Refine" which is also common property is implemented and well, it is also must be called in all features. So this is fine I suppose, maybe an idea for later refactor. I'm marking this as resolved as it is not a PR that should do a major overhaul of how features are implemented.

Reply:
@kadet1090 I found a solution by override `recompute`

Reply:
It doesn't work as expected, I can edit a existing feature but no more create a new one.
Fall back to working previous but still work on this solution.